lib/Dialect/AIE/Transforms/AIEObjectFifoStatefulTransform.cpp

```diff
@@ -1000,10 +1000,12 @@ struct AIEObjectFifoStatefulTransformPass
     createFifoOps.insert(createFifoOps.end(), range.begin(), range.end());
     for (auto createOp : createFifoOps) {
       std::vector<ObjectFifoCreateOp> splitConsumerFifos;
-      int consumerIndex = 0;
+      size_t consumerIndex = 0;
       int consumerDepth = createOp.size();
       ArrayRef<BDDimLayoutArrayAttr> consumerDims =
           createOp.getDimensionsFromStreamPerConsumer();
+      assert(consumerDims.size() >= 1 &&
+             "At least one consumer `BDDimLayoutArrayAttr` expected");
```

Comment:
It should be possible to have a broadcast without any data layout transformations.

Reply:
In my opinion, after running `parseObjectFifoConsumerTiles()` we should actually assert `assert(consumerDims.size() == createOp.getConsumerTiles().size())`.